### DISCUSSION TITLE
How to install latest FW/SW on old NSPANEL Pro with V1.3.3 FW and ADB enabled?

### DISCUSSION THREAD

**Discussion by syoma755** [seaky/nspanel_pro_tools_apk#161] [General]
Hi there,

Not sure if it is right forum and tool, so please excuse for dumb questions.

I've got my NSPANEL Pro back in 2022 and since it's firmware was not at the state I wished (missing Node-RED support) I enabled ADB, played a little bit, and then due to lack of time stopped working with it.

Now I've checked and found that NSPANEL supports Node-Red natively and wanted to install latest FW, but off course failed, since ADB enabled devices are not updateable.

So my question to this community - with nspanel_pro_tools can I install latest orginal Sonoff FW (I need at least 1.7 or later), and how does it work?

Thanks
Artem

**Comment by houtacheng** [comment]
Use this:
https://github.com/seaky/nspanel_pro_roottool_apk


3.5
https://github.com/seaky/nspanel_pro_tools_apk/discussions/134
